src/Core/FeatureFlag/FeatureFlagsModifier.php

```diff
@@ -0,0 +1,139 @@
+ * through the Back Office interface. It fetches the data needed for the Back Office interface
+ * and handles payloads from the Back Office page, replicating the changes on the data model.
+ */
+class FeatureFlagsModifier implements DataConfigurationInterface
+{
+    /** @var EntityManagerInterface */
+    private $doctrineEntityManager;
+
+    /** @var TranslatorInterface */
+    private $translator;
+
+    /**
+     * @param EntityManagerInterface $doctrineEntityManager
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(EntityManagerInterface $doctrineEntityManager, TranslatorInterface $translator)
+    {
+        $this->doctrineEntityManager = $doctrineEntityManager;
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getConfiguration(): array
+    {
+        $configuration = [];
+
+        foreach ($this->getAllFeatureFlags() as $id => $featureFlag) {
+            $configuration[$featureFlag->getName()] = $featureFlag->getState();
+        }
+
+        return $configuration;
+    }
+
+    /**
+     * @return FeatureFlag[] $allFlags
+     */
+    public function getAllFeatureFlags(): array
+    {
+        return $this->doctrineEntityManager->getRepository('PrestaShopBundle:FeatureFlag')->findAll();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function updateConfiguration(array $configuration): array
+    {
+        /** @var array<string, boolean> $configuration */
+        foreach ($configuration as $flagName => $flagState) {
+            $this->validateConfigurationRowType($flagName, $flagState);
+
+            $featureFlag = $this->getOneFeatureFlagByName($flagName);
+
+            if ($flagState) {
+                $featureFlag->enable();
+            } else {
+                $featureFlag->disable();
+            }
+        }
+
+        $this->doctrineEntityManager->flush();
+
+        return [];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function validateConfiguration(array $configuration): bool
+    {
+        /** @var array<string, boolean> $configuration */
+        foreach ($configuration as $flagName => $flagState) {
+            $this->validateConfigurationRowType($flagName, $flagState);
+
+            if (null === $this->getOneFeatureFlagByName($flagName)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * @param string $featureFlagName
+     *
+     * @return FeatureFlag|null return null if feature flag cannot be found
+     */
+    public function getOneFeatureFlagByName(string $featureFlagName)
+    {
+        return $this->doctrineEntityManager->getRepository('PrestaShopBundle:FeatureFlag')->findOneBy(['name' => $featureFlagName]);
+    }
+
+    /**
+     * @param string $featureFlagName
+     * @param bool $flagState
+     */
+    private function validateConfigurationRowType(string $featureFlagName, bool $flagState): void
```

Comment:
If it's not needed, you can remove this method :thinking:

Reply:
It's useful ! Actually what I wanted to do is a function to assert that `$featureFlagName` is a string and `$flagState` is a boolean.

Before PHP7 I would have done
```
    private function validateConfigurationRowType($featureFlagName, $flagState): void
    {
        if (!is_string(featureFlagName)) { throw new \Exception(); }
        
        if (!is_bool(flagState)) { throw new \Exception(); }
    }
```

But when we use strict types, I dont even need to write the body of the function 😄  type checking happens at function level

Reply:
It's for readability and performance.
Using `!is_string` is quite faster than calling a method and then check if parameters are properly defined, then returning void.

```
<?php

declare(strict_types=1);

class A
{
    public function validateFields(array $data): void
    {
        $startTime = microtime(TRUE);
        for ($i = 0; $i < 10000; $i++) {
            $this->test($data['string'], $data['bool']);
        }
        $endTime = microtime(TRUE);

        echo $endTime - $startTime;
        echo PHP_EOL;


        $startTime = microtime(TRUE);
        for ($i = 0; $i < 10000; $i++) {
            if (!is_string($data['string']) || !is_bool($data['bool'])) {
            }
        }
        $endTime = microtime(TRUE);

        echo $endTime - $startTime;
        echo PHP_EOL;
    }

    public function test(string $a, bool $b): void
    {
        //Nothing to do
    }
}

$class = new A();
$class->validateFields(
    [
        'string' => 'test',
        'bool' => false,
    ]
);
```
The second one is 5x faster than the first one:
```
First: 0.0026290416717529
Second: 0.00058603286743164

First: 0.0038280487060547
Second: 0.00078487396240234

First: 0.0039999485015869
Second: 0.00078701972961426

First: 0.0039198398590088
Second: 0.00079488754272461

First: 0.0037069320678711
Second: 0.0008091926574707

First: 0.0051801204681396
Second: 0.00058197975158691

First: 0.0027329921722412
Second: 0.00062394142150879

First: 0.0034170150756836
Second: 0.00068402290344238

First: First: 0.0039041042327881
Second: 0.00078892707824707
```

You're also not controlling the exception which will always be a TypeError but it's not always a problem.


The second point that can be painful is the number of arguments and useless methods you will have to use in order to perform this kind of methods.
```
public function validateFieldsAreValids(string, int, bool, array, int, bool, string, ..)
{
 
}
```

Reply:
Thanks for the benchmark. Very interesting.

Reply:
I also agree that this empty function looks ugly 🤔 If you really need some strong type validation, then it would be cleaner to use strong types as inputs of your `updateConfiguration` method Instead of relying on an `array` you can create some `FeatureFlag` VOs with a string and a boolean, it will be easier to understand, give you type assertions and help auto completion 😉

Reply:
@PierreRambaud Does this mean strict type hinting has a significant speed cost?

@jolelievre The signature of `updateConfiguration()` is frozen by the interface so I cannot use a VO.

Reply:
Fixed in https://github.com/PrestaShop/PrestaShop/pull/23703/commits/410c7ff73ba8a82c41dbed94cb5403810fe27d8e